### PR TITLE
added auth & register endpoint

### PR DESCRIPTION
**Pull request by Niceley**
added auth & register endpoint

### CODE REVIEW

File: src/controller/auth.js [hunk 1]
```diff
@@ -8,69 +8,64 @@ module.exports = {
         try {
             verifyUser(req.body);
             const { firstname, lastname, email, password } = req.body;
+
+            const existingUser = await UserModel.findOne({ email });
+            if (existingUser) {
+                return res.status(400).send({
+                    message: "Email already exists",
```
Comment (Orden14) [inline src/controller/auth.js:15]:
```suggestion
                    message: "Un compte avec cet email existe déjà",
```

File: src/app.js [hunk 2]
```diff
@@ -8,23 +8,32 @@ const swaggerUI = require("swagger-ui-express");
...
 
 app.use(express.json());
 
 const swaggerOptions = {
     swaggerDefinition: {
         openapi: "3.0.0",
         info: {
-            title: "MyCv API",
-            version: "1.0.0",
+            title: "Book API",
```
Comment (Niceley) [inline src/app.js:24]: MyCV API

File: src/controller/auth.js [hunk 3]
```diff
@@ -8,69 +8,64 @@ module.exports = {
...
 
-            newUser.save();
+            await newUser.save();
             res.status(201).send({
                 id: newUser._id,
                 lastname: newUser.lastname,
                 firstname: newUser.firstname,
                 email: newUser.email,
             });
+            console.log("User registered successfully");
```
Comment (Orden14) [inline src/controller/auth.js:34]: enlever le console log

File: src/controller/auth.js [hunk 4]
```diff
@@ -8,69 +8,64 @@ module.exports = {
...
         } catch (error) {
-            res.send({
+            res.status(500).send({
                 message: error.message || "Cannot register User",
```
Comment (Orden14) [inline src/controller/auth.js:37]:
```suggestion
                message: error.message || "Impossibe d'enregistrer l'utilisateur",
```

File: src/app.js (hunk 2 above)
Comment (Orden14) [inline src/app.js:24]:
```suggestion
            title: "MyCv API",
```

File: src/app.js [hunk 5]
```diff
@@ -8,23 +8,32 @@ const swaggerUI = require("swagger-ui-express");
...
+            version: "1.1.0",
         },
     },
-    apis: ["./routes/*.js"],
+    apis: ["./src/routes/*.js"],
 };
 
 const swaggerDocs = swaggerJsDoc(swaggerOptions);
 
+console.log(swaggerDocs);
```
Comment (Orden14) [inline src/app.js:33]: enlever le console.log

**Review by Orden14 — APPROVED** [review]
ready to merge

**Pull request merged**